src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java

```diff
@@ -1024,4 +1024,13 @@ default void onBlockExploded(BlockState state, World world, BlockPos pos, Explos
         world.setBlockState(pos, Blocks.AIR.getDefaultState(), 3);
         getBlock().onExplosionDestroy(world, pos, explosion);
     }
+
+    /**
+     * Determines if this block's collision box should be treated as though it can extend above its block space.
+     * Use this to replicate fence and wall behavior.
+     */
+    default boolean collisionExtendsVertically(BlockState state, IBlockReader world, BlockPos pos, Entity collidingEntity)
+    {
+        return false;
```

Comment:
I'd argue we should change it to use the two contains checks on the tags that vanilla always did, and move remove the check from the original method, so you can override this to false even if it is in the fences or wall tags. Just to give it even more flexibility...